### DISCUSSION TITLE
Issue? No Discussion Board I guess! Just need help sorting out my unlinked setup

### DISCUSSION THREAD

**Discussion by BuildBackBuehler** [jason0x43/dotfiles#3] [General]
So first off, the dotfile function is a godsend, thank you for that and for just sharing your elegant/sophisticated setup. Honestly surprised it doesn't have a lot more stars/forks, though I only got lucky having studied up on what the latest & greatest is for a terminal and your dotfile shows up. 

And last thing before I get into the SOS, I'm just someone who is leaning into programming/coding to really bulk up and refine what I (intend to) offer in the form of marketing/advertising services for small businesses. I've also been messing around with ML for the past couple years, just dabbling in some style transfer type stuff locally on my rig once-in-a-blue-moon. 

My formal training amounts to Python 101 & messing around with ML, more recently, my Synology NAS and (attempting to) build some premium quality websites (next.js). So a year ago I had managed to setup my Mac (ARM based, M1) properly, but with everything else, I must've absolutely mangled the symlinking and/or editing the wrong files (my /User/[name]/[git/npm/nvm/zsh/kubernetes etc.]). When you're just brute force tweaking code without the foundation, you just work with what's in front of you, at least in my case 😅.  

But I've been working on my dotfiles for 2-3 days now and feeling not so 101 beginner. Of course I just figured that I had mangled the actual code itself because it was only a few things not working, finicky stuff like zsh-autocomplete, or my p9k config switching for kubernetes...and having like 3 different, redundant node_module directories. 

I apologize because it'd be so much easier to just talk something like this out, but now I replenished/re-upped my dotfiles by more-or-less mirroring yours. I made some minor tweaks but I also don't *actually* know what's best and why that is, I just knew y'all knew your "sh!p" – so I'm open to any suggestions to make my setup the "best" while still being user-friendly/idiot proof/easy to learn. 

Andddd so the actual issue:

I've been editing my .zshrc/zprofile/zshenv from inside /User/[name]/ and manually messing with my setup that I had in a "[.gitrepodirectory]" – though at some point a ~/.local/[bin/share/state] folder also came into play...and ~/[.config/.cache] came into play. 

Fast forward and this is what I've got. 

<img width="732" alt="image" src="https://user-images.githubusercontent.com/116780517/237024134-46dd1c56-6c4c-4ed6-87b9-7d640bcc4e78.png">

When I was catching on to the lack of linkage, I started w/ making aliases of some ~/.dotfiles directories as you can see (not symlinked, just created it in the finder command menu...minus bin)

<img width="606" alt="image" src="https://user-images.githubusercontent.com/116780517/237022912-fc82914f-11f8-478d-a60c-0e22c0bfd602.png">

My origin brew files...just in case it helps

<img width="832" alt="image" src="https://user-images.githubusercontent.com/116780517/237025327-65f59b31-a5fa-430f-a674-45e12347c60a.png">

And when I saw this, here at 0200, I figured duck it, this isn't going to fix jack ship...

<img width="545" alt="image" src="https://user-images.githubusercontent.com/116780517/237026075-8ec6ece3-aad8-420e-8628-900ef2c47676.png">

As you can see total = 0...

To potentially further complicate things, and maybe I just need to full nuke my system, I had been using a multi-user system, but when I was having issues with insecure folder compdef errors, I changed my system to single-user (though there's my User folder "zack" & a presumably derelict "shared" folder). 

Hate for this to be so long, but I guess its the nature of the beast, based on all the walls of text from strugglers' with similar issues. Hope to hear back from you soon, thanks again!

**Comment by BuildBackBuehler** [comment]
Another potential issue is I recently updated to a broken Ventura Beta 13.4 which has messed up my 3rd party networking at the very least, but just saw someone who said all their permissions for accessibility and full disk access had been wiped. 


```shell
brew config
HOMEBREW_VERSION: 4.0.17-30-g82dc1d2
ORIGIN: https://github.com/Homebrew/brew
HEAD: 82dc1d25ef23ef9a23b9f32e4b22d3d79d291f35
Last commit: 7 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: c70cee274c84e86069a22c449933725ab692b479
Core tap last commit: 11 hours ago
Core tap branch: master
Core tap JSON: 09 May 06:02 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_BAT_THEME: wezterm
HOMEBREW_BOOTSNAP: set
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.40.1 => /opt/homebrew/bin/git
Curl: 7.88.1 => /usr/bin/curl
macOS: 13.4-arm64
CLT: 14.3.0.0.1.1679647830
Xcode: N/A
Rosetta 2: false
```

**Comment by jason0x43** [comment]
> Issue? No Discussion Board I guess!

I hadn't really considered that anyone might want to discuss my dotfiles. You're right, a discussion definitely makes more sense than an issue for this, so I enabled those. 🙂

So...what exactly is the issue you're running into? In reading through your post, nothing is really jumping out at me as "bad".

**Comment by BuildBackBuehler** [comment]
Yeah well unfortunately  any sort of "how to do ___" or "what is the best setup for ___" ends up with top results from big sites with the sole goal of push of their solution (platform or product) or amateurs in coding (et al.), but experts in SEO with the sole goal of monetization. Thus one is left with generic or simplistic instructions. There's Overstackflow for discussing coding stuff but find it rather uninviting, have only visited whenever a thread is in my search results, lurking for solutions.

I imagine there are a lot of people like myself who are interested in talking with someone/people who are knowledgable while learning something like this. You should definitely write more practical articles and drop a buymeacoffee-type link at the bottom of your articles! 

Y'anyways, well, now that you mention it, I suppose because I had my laptop setup for multiple users, that was the only way Homebrew could work (symlinked to each users' home directories). I suppose that's not necessary but seems the rule of thumb is to have redundancies and to keep distinctive environments that don't rely on the same packages (for ex., if I screwed up my NAS' zsh environment while tinkering with Kubernetes, it'd be catastrophic if I then needed to re-setup my local zsh env. and my website building env.). How do you suggest I avoid such a fate? Or will just using direnv./ASDF take care of that? Thanks!

**Comment by jason0x43** [comment]
Hmmm...I'm still not entirely clear what the issue is. Is it that you want to have project-specific shell environments?

Project-specific settings, ASDF + direnv, or the newer [rtx](https://github.com/jdxcode/rtx), can be very helpful. These will let you install different versions of tools (Nodejs, Ruby, etc.) and set environment variables on a per-directory basis. Direnv will even let you run shell scripts when entering a project directory.
